tuf/utils/x509.go

```diff
@@ -443,6 +454,9 @@ func EncryptPrivateKey(key data.PrivateKey, role, passphrase string) ([]byte, er
 		return nil, fmt.Errorf("unable to encrypt key - invalid PEM file produced")
 	}
 	encryptedPEMBlock.Headers["role"] = role
+	if gun != "" {
```

Comment:
nit: Is there a reason why we couldn't handle empty guns like we do for empty roles?  Since we're doing this special gun handling, should the `"role"` header have an if block as well in case delegation keys do not have roles?